### PR TITLE
Uncomment AdminInterface::isCurrentRoute method for next major relase

### PR DESCRIPTION
**Pull request by Rivsen**
Related PR is #3748

### CODE REVIEW

File: Resources/doc/cookbook/recipe_dynamic_form_modification.rst [hunk 1]
```diff
@@ -45,5 +45,8 @@ Then, you should be able to dynamically add needed fields to the form:
             if ($this->isCurrentRoute('create')) {
                 $formMapper->add('name', 'text');
             }
+
+            //$this->isCurrentRoute('create'); // is create page?
```
Comment (core23) [inline Resources/doc/cookbook/recipe_dynamic_form_modification.rst]: This is the wrong place for the snippet. Why should we call the methods here?

Comment (Rivsen) [inline Resources/doc/cookbook/recipe_dynamic_form_modification.rst]: I'm sorry. It's just some examples here, where the code snippet should be placed?

**Comment by Soullivaneuh** [comment]
@Rivsen Could you please rebase? Also changelog and upgrade format is changed.

**Comment by Rivsen** [comment]
@Soullivaneuh I rebased the master, is this correct?

File: UPGRADE-4.0.md [hunk 2]
```diff
@@ -1,2 +1,5 @@
 UPGRADE FROM 3.x to 4.0
 =======================
+
+## Admin
+The ``isCurrentRoute()`` method was added to the AdminInterface, You can easily determine which action is the current request.
```
Comment (Soullivaneuh) [inline UPGRADE-4.0.md]: > to the AdminInterface, You

I think `,` should be a `.`. :-)

Comment (Rivsen) [inline UPGRADE-4.0.md]: Thanks, fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 ### Added
+- Add the ``isCurrentRoute`` method to the AdminInterface
```
Comment (Soullivaneuh) [inline CHANGELOG.md]: * Only one backtick please.
* Inline code fore `AdminInterface`

File: UPGRADE-4.0.md [hunk 4]
```diff
@@ -1,2 +1,5 @@
 UPGRADE FROM 3.x to 4.0
 =======================
+
+## Admin
+The ``isCurrentRoute()`` method was added to the AdminInterface. You can easily determine which action is the current request.
```
Comment (Soullivaneuh) [inline UPGRADE-4.0.md]: * Only one backtick
* Inline code for `AdminInterface`
* Remove "You can easily determine..." setence
* Rebase again against master and complete this part: https://github.com/sonata-project/SonataAdminBundle/blob/master/UPGRADE-4.0.md#admin

**Comment by Rivsen** [comment]
Yes, I will update this PR tonight.

On Thu, May 19, 2016, 20:08 Sonata CI <notifications@github.com> wrote:

> Could you please rebase your PR and fix merge conflicts?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/sonata-project/SonataAdminBundle/pull/3762#issuecomment-220305334>
>
-- 

Send from my Nexus 5.

**Comment by Rivsen** [comment]
@Soullivaneuh I rebased the master and fixed conflicat, please check. Thank you. :-)

File: Resources/doc/cookbook/recipe_dynamic_form_modification.rst (hunk 1 above)
Comment (greg0ire) [inline Resources/doc/cookbook/recipe_dynamic_form_modification.rst]: ping @core23

Comment (core23) [inline Resources/doc/cookbook/recipe_dynamic_form_modification.rst]: I think we can remove this lines here.

Comment (Soullivaneuh) [inline Resources/doc/cookbook/recipe_dynamic_form_modification.rst]: Let's do this. Documentation improve can be done on another PR if wanted.

cc @Rivsen

**Comment by Soullivaneuh** [comment]
Sorry for this, could you rebase again? Will try to merge it ASAP.

**Comment by Soullivaneuh** [comment]
@sonata-project/contributors LGTM

File: Resources/doc/cookbook/recipe_dynamic_form_modification.rst [hunk 5]
```diff
@@ -45,5 +45,7 @@ Then, you should be able to dynamically add needed fields to the form:
             if ($this->isCurrentRoute('create')) {
                 $formMapper->add('name', 'text');
             }
+
+            //$this->isCurrentRoute('edit', 'some.admin.code'); // is some.admin.code admin's edit page?
```
Comment (core23) [inline Resources/doc/cookbook/recipe_dynamic_form_modification.rst]: Maybe add another example here?

```php
             if ($this->isCurrentRoute('edit', 'acme.demo.admin.code')) {
                 $formMapper->add('foo', 'text');
             }
```

Comment (Soullivaneuh) [inline Resources/doc/cookbook/recipe_dynamic_form_modification.rst]: Indeed, your example if more comprehensible.

@Rivsen Could you please update this part? Thanks.

**Comment by Soullivaneuh** [comment]
Can be merge once CI will be green.

Thank you @Rivsen!

**Pull request merged**